tests/regression/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+
+
+
+// Function to create default config with specified parameters
+pub fn set_config(
+    cipher_prefs: &security::Policy,
+    keypair: CertKeyPair
+) -> Result<s2n_tls::config::Config, Error> {
+    let mut builder = Builder::new();
+    builder
+        .set_security_policy(cipher_prefs)
+        .expect("Unable to set config cipher preferences");
+    builder
+        .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
+        .expect("Unable to set a host verify callback.");
+    builder
+        .load_pem(keypair.cert(), keypair.key())
+        .expect("Unable to load cert/pem");
+    builder.trust_pem(keypair.cert()).expect("load cert pem");
+    builder.build()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crabgrind as cg;
+    use s2n_tls::testing::TestPair;
+    use std::env;
+    use std::process::Command;
+
+    const COST: u64 = 1_000_000; //configurable threshold for regression
+    // environment variable to determine whether to run under valgrind or solely test functionality
+    fn is_running_under_valgrind() -> bool {
+        env::var("VALGRIND").is_ok()
+    }
+    //test to create new config, set security policy, host_callback information, load/trust certs, and build config
+    #[test]
+    fn test_set_config() -> Result<(), s2n_tls::error::Error> {
+        if !is_running_under_valgrind() {
+            cg::cachegrind::stop_instrumentation();
+            cg::cachegrind::start_instrumentation();
+            let keypair_rsa = CertKeyPair::default();
+            let _config = set_config(&security::DEFAULT_TLS13, keypair_rsa).expect("Failed to build config");
+            return Ok(());
+        }
+        run_valgrind_test("test_set_config");
+        Ok(())
+    }
+
+    #[test]
+    fn test_rsa_handshake() -> Result<(), s2n_tls::error::Error> {
+        if !is_running_under_valgrind() {
+            cg::cachegrind::stop_instrumentation();
+            // Example usage with RSA keypair (default)
+            let keypair_rsa = CertKeyPair::default();
+            let config = set_config(&security::DEFAULT_TLS13, keypair_rsa)?;
+            // Create a pair (client + server) using that config, start handshake measurement
+            let mut pair = TestPair::from_config(&config);
+            // Assert a successful handshake
+            cg::cachegrind::start_instrumentation();
+            assert!(pair.handshake().is_ok());
+            cg::cachegrind::stop_instrumentation();
+            return Ok(());
+        }
+        run_valgrind_test("test_rsa_handshake");
+        Ok(())
+    }
+    // function to run specified test using valgrind
+    fn run_valgrind_test(test_name: &str) {
+        let exe_path = std::env::args().next().unwrap();
+        let output_file = format!("cachegrind_{}.out", test_name);
+        let valgrind_command = format!(
+            "valgrind --tool=cachegrind --cachegrind-out-file={} {} {}",
+            output_file, exe_path, test_name
+        );
```

Comment:
The `Command` should implement debug: https://doc.rust-lang.org/std/process/struct.Command.html#impl-Debug-for-Command so this should go through that instead